ext/serializer.c

```diff
@@ -1256,6 +1256,25 @@ void ddtrace_serialize_span_to_array(ddtrace_span_data *span, zval *array) {
         }
 
         add_assoc_zval(el, "service", &prop_service_as_string);
+
+        // Add _dd.base_service if service name differs from mapped root service name
+        zval *prop_root_service = &span->root->property_service;
+        ZVAL_DEREF(prop_root_service);
+        zval prop_root_service_as_string;
+        ddtrace_convert_to_string(&prop_root_service_as_string, prop_root_service);
+
+        zval *new_root_name = zend_hash_find(service_mappings, Z_STR(prop_root_service_as_string));
+        if (new_root_name) {
+            zend_string_release(Z_STR(prop_root_service_as_string));
+            ZVAL_COPY(&prop_root_service_as_string, new_root_name);
+        }
```

Comment:
This persists the mapping on the root service the first time a span is serialized.
Especially in future, when we gain partial trace flushing, this will be a quite the spooky action at a distance.

We anyway have to do the lookup each time (to check whether the root span service name is not an unmapped one), so doing that doesn't save us anything. I don't think we should copy the mapped service name back to the root span.